### DISCUSSION TITLE
Does Mixin need a Docker image?

### DISCUSSION THREAD

**Discussion by LixvYang** [MixinNetwork/mixin#166] [Ideas]
Key Benefits of Packaging Mixin Projects into Docker

Environment isolation. Through Docker, Mixin can provide an isolated running environment, avoiding conflicts between environment variables or other impacts. Using Docker eliminates the need to consider compatibility issues, different Mixin instances can easily coexist.

Simpler deployment. Through a Docker image representing Mixin, anyone can easily deploy and use the Mixin system without additional environment setup. This significantly reduces barriers to entry and increases the usability of Mixin.

Better scope control. Using Docker Toolbox can easily set up an isolated Mixin development environment, avoid direct operation on the host machine, and easily rollback environments. Using Docker Swarm can also deploy clusters to separate user access to underlying environments.

Enhanced security. The Docker environment can further isolate the Mixin runtime environment through sandboxing, increasing security barriers. It requires special authorization to impact the underlying operating system, effectively preventing attacks against Mixin.

Stronger portability. Proven stable running in Docker environments can easily run Mixin on other machines. Not dependent on a specific platform or operating system, Mixin can spread efficiently anywhere.

In summary, the primary benefits of pushing Mixin to Docker are to provide a more isolated, simple, reliable and efficient running environment, easily facilitating operational smoothness and increased Mixin delivery frequency. I hope these analyses can help the Mixin team make more reasonable judgments in similar scenarios!

**Comment by cedricfung** [comment]
Since Mixin is a very simple go binary, even without cgo dependencies, it's very portable to run on almost all platforms that Go supports. So deployment and portability are not the case here. 

I don't understand the security of development either, because Mixin doesn't even depend on any third party services, just a single contained binary to run anywhere without the need to install or configure anything.

**Reply by LixvYang** [reply]
Well you are right, looks like mixin no need for a mirror like service, because it's very simple.
